pytket/pytket/wasm/wasm.py

```diff
@@ -236,8 +256,52 @@ def check_function(
         :type number_of_returns: int
         :return: true if the signature and the name of the function is correct"""
 
-        return not self._check_file or (
+        return not self.checked or (
             (function_name in self._functions)
             and (self._functions[function_name][0] == number_of_parameters)
             and (self._functions[function_name][1] == number_of_returns)
         )
+
+    @cached_property
+    def functions(self) -> dict[str, tuple[int, int]]:
+        """Retrieve the names of functions with the number of input and out arguments.
+
+        Will check the module if it has not been checked already.
+        """
+        if not self.checked:
```

Comment:
Possibly this block could be moved into check() itself but it seemed a bit more explicit to do it here, open to suggestions.

Reply:
I am not fully sure that it is good to run the checks here (and on other places) secretly ignoring that the user first requested the file not to be checked. Should we at least give a warning that this is done?

Reply:
I'm not really sure I understand the use-case behind not checking the module, is the intention to save compute, to avoid erroneous checks or to allow submission of broken code?

Reply:
The idea is to avoid erroneous checks and allow the submission of broken code.